### PR TITLE
HPCC-30217 Sorting paged grids resets column widths

### PR DESCRIPTION
**Pull request by GordonSmith**
Remove refresh flicker

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: esp/src/src-react/components/controls/Grid.tsx [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+    return <DetailsList
+        compact={true}
+        items={items}
+        columns={fluentColumns}
+        setKey="set"
+        layoutMode={DetailsListLayoutMode.justified}
+        selection={selectionHandler}
+        isSelectedOnFocus={false}
+        selectionPreservedOnEmptyClick={true}
+        // onItemInvoked={this._onItemInvoked}
```
Comment (jeclrsg) [inline esp/src/src-react/components/controls/Grid.tsx:174]: commented code

File: esp/src/src-react/components/controls/Grid.tsx [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+                { key: 500, text: "500" },
+                { key: 1000, text: "1000" }
+            ]} selectedKey={pageSize} onChange={dropdownChange} />
+        </Stack.Item>
+    </Stack>;
+};
+
+// import { useWhatChanged } from "@simbathesailor/use-what-changed";
+
+export function useXXX({
```
Comment (jeclrsg) [inline esp/src/src-react/components/controls/Grid.tsx:344]: this was something for testing?

File: esp/src/src-react/components/controls/Grid.tsx [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+        });
+        storeQuery.then(items => {
+            setItems(items);
+            setSelection(selectionHandler.getSelection());
+        });
+    }, [count, selectionHandler, start, store], [query, sorted]);
+
+    React.useEffect(() => {
+        refreshTable(refresh.clear);
+    }, [refreshTable, refresh.value, refresh.clear]);
```
Comment (jeclrsg) [inline esp/src/src-react/components/controls/Grid.tsx:120]: refresh.value is not needed?

**Review by jeclrsg — CHANGES_REQUESTED** [review]
@GordonSmith looks good. Just saw a few lines that were for testing? Commented code and such.

File: esp/src/src-react/components/controls/Grid.tsx (hunk 2 above)
Comment (GordonSmith) [reply esp/src/src-react/components/controls/Grid.tsx:344]: No it should be removed.

File: esp/src/src-react/components/controls/Grid.tsx (hunk 3 above)
Comment (GordonSmith) [reply esp/src/src-react/components/controls/Grid.tsx:120]: It is needed - when you click the refresh button on the page, it increments refresh.value under the covers which triggers the refresh - I should probably add a comment.

File: esp/src/src-react/components/Queries.tsx [hunk 4]
```diff
@@ -96,89 +96,91 @@ export const Queries: React.FunctionComponent<QueriesProps> = ({
...
-                }, [])
+                }
             },
             ErrorCount: {
                 headerIcon: "Warning",
                 headerTooltip: nlsHPCC.ErrorWarnings,
                 width: 16,
                 sortable: false,
-                formatter: React.useCallback(function (error) {
+                formatter: function (error) {
```
Comment (GordonSmith) [inline esp/src/src-react/components/Queries.tsx:111]: Should use arrow function

File: esp/src/src-react/components/Queries.tsx [hunk 5]
```diff
@@ -96,89 +96,91 @@ export const Queries: React.FunctionComponent<QueriesProps> = ({
...
-                }, [])
+                }
             },
             MixedNodeStates: {
                 headerIcon: "Error",
                 headerTooltip: nlsHPCC.MixedNodeStates,
                 width: 16,
                 sortable: false,
-                formatter: React.useCallback(function (mixed) {
+                formatter: function (mixed) {
```
Comment (GordonSmith) [inline esp/src/src-react/components/Queries.tsx:123]: Should use arrow function

File: esp/src/src-react/components/Workunits.tsx [hunk 6]
```diff
@@ -124,23 +124,23 @@ export const Workunits: React.FunctionComponent<WorkunitsProps> = ({
                 headerTooltip: nlsHPCC.Protected,
                 width: 16,
                 sortable: true,
-                formatter: React.useCallback(function (_protected) {
+                formatter: function (_protected) {
```
Comment (GordonSmith) [inline esp/src/src-react/components/Workunits.tsx:127]: Should use arrow function

File: esp/src/src-react/components/Workunits.tsx [hunk 7]
```diff
@@ -124,23 +124,23 @@ export const Workunits: React.FunctionComponent<WorkunitsProps> = ({
...
                         return <Icon iconName="LockSolid" />;
                     }
                     return "";
-                }, [])
+                }
             },
             Wuid: {
                 label: nlsHPCC.WUID, width: 120,
-                formatter: React.useCallback(function (Wuid, row) {
+                formatter: function (Wuid, row) {
```
Comment (GordonSmith) [inline esp/src/src-react/components/Workunits.tsx:136]: Should use arrow function

File: esp/src/src-react/components/controls/Grid.tsx (hunk 1 above)
Comment (GordonSmith) [reply esp/src/src-react/components/controls/Grid.tsx:174]: Removed.

File: esp/src/src-react/components/controls/Grid.tsx (hunk 2 above)
Comment (GordonSmith) [reply esp/src/src-react/components/controls/Grid.tsx:344]: Removed

File: esp/src/src-react/components/controls/Grid.tsx (hunk 3 above)
Comment (GordonSmith) [reply esp/src/src-react/components/controls/Grid.tsx:120]: Updated with NOP reference and comment

**Review by jeclrsg — APPROVED** [review]

**Pull request merged**